README.md

```diff
@@ -307,6 +308,59 @@ Let’s create two new functions that let us deposit and withdraw liquidity. How
 
 ```
 
+<details markdown='2'><summary>🤔 Why are we adding 1 wei in `tokenDeposit` calculation ?</summary>
```

Comment:
Thanks for this PR! I overall think that this is on the right track, tbh I need to shake off the rust on this discussion point / edge case. Since you have already taken the initiative though with this addition, I would take a look at the conversation around the origin of this piece of code [here](https://t.me/c/1655715571/106) if you have not yet. Maksim summarizes it nicely to what I recall being the reason we decided to implement it (edge-cases). 

Perhaps link to this spot in the telegram chat as the original spot of discussing adding `add(1)` to the solutions code. As well, I wonder if this is really also important for actual DEX code like uniswap - our DEX is a simple rendition, whereas uniswap actually has `price()` functions possibly incorporated with this `add(1)` aspect instead of us just using it in the `deposit()`. If you have time, check out uniswap to see if they just have it in their `deposit()` function for liquidity, or if they have it used in price too or elsewhere. Don't spend an insane amount of time on it I'd say.

The point is to just see if it is incorporated more-so in uniswap, and if it is, to note that in the README and explain that a lot of the inspiration was pulled from uniswap, like this `add(1)` edge-case implementation code. As well that our DEX is a simpler version, and in uniswap this edge-case is actually a big deal but out of scope for this task so not to focus on it much more :)

wdyt?

Reply:
Thanks so much, @umphams 🙌 Added the link to the discussion in the solution code. 
I went through the contracts of Uniswap V1 and V2  and here are my findings : 
## V1
### Price calculation : 
V1 has two functions for price calculation : 
* [getInputPrice](https://github.com/PhABC/uniswap-solidity/blob/2434367a66b6091db7b808ed91e7ade61fad6f7d/contracts/uniswap/UniswapExchange.sol#L70) - Given inputTokens returns you amount of output tokens you will receive. our `price` function . 
* [getOutputPrice](https://github.com/PhABC/uniswap-solidity/blob/2434367a66b6091db7b808ed91e7ade61fad6f7d/contracts/uniswap/UniswapExchange.sol#L85) - GIven the amount of outputTokens you want returns you the amounts of inputToken you should give. **This `price` function incorporates `add(1)`** . 

[getOutputPrice](https://github.com/PhABC/uniswap-solidity/blob/2434367a66b6091db7b808ed91e7ade61fad6f7d/contracts/uniswap/UniswapExchange.sol#L85) is used by all the `swap` functions  in which you mention the amount of `outputToken` you want. 
eg : [ethToTokenTransferOutput](https://github.com/PhABC/uniswap-solidity/blob/2434367a66b6091db7b808ed91e7ade61fad6f7d/contracts/uniswap/UniswapExchange.sol#L159) and similar others. 

### addLiquidity (deposit function) : 
It also incorporates `.add(1)` ->   [addLiquidity](https://github.com/PhABC/uniswap-solidity/blob/2434367a66b6091db7b808ed91e7ade61fad6f7d/contracts/uniswap/UniswapExchange.sol#L560) . 

## V2 
### addLiquidity (deposit function) : 
Under the hood [addLiquidity](https://github.com/Uniswap/v2-periphery/blob/0335e8f7e1bd1e8d8329fd300aea2ef2f36dd19f/contracts/UniswapV2Router02.sol#L33) uses [quote](https://github.com/Uniswap/v2-periphery/blob/0335e8f7e1bd1e8d8329fd300aea2ef2f36dd19f/contracts/libraries/UniswapV2Library.sol#L36) which do not uses `add(1)` because `addLiquidity` in v2 is implemented a bit differently as it accepts both the token at the time of deposit . 

### Price calculation : 
Similar to V1, V2 also has similar two function : 
* [getAmountOut](https://github.com/Uniswap/v2-periphery/blob/0335e8f7e1bd1e8d8329fd300aea2ef2f36dd19f/contracts/libraries/UniswapV2Library.sol#L43)
 * [getAmountIn](https://github.com/Uniswap/v2-periphery/blob/0335e8f7e1bd1e8d8329fd300aea2ef2f36dd19f/contracts/libraries/UniswapV2Library.sol#L53) which incorporates `.add(1)`

Found this nice [explanation](https://www.reddit.com/r/UniSwap/comments/rjzkcj/why_does_it_have_add1_uniswapv2library_getamountin/) for `.add(1)` for `getAmountIn` function